src/transforms/sample/transform.rs

```diff
@@ -8,31 +10,36 @@ use crate::{
     transforms::{FunctionTransform, OutputBuffer},
 };
 
+const DEFAULT_GROUP_NAME: &str = "default";
+
 #[derive(Clone)]
 pub struct Sample {
     name: String,
     rate: u64,
     key_field: Option<String>,
+    group_by: Option<String>,
     exclude: Option<Condition>,
-    count: u64,
+    counter: HashMap<String, u64>,
```

Comment:
```suggestion
    counter: HashMap<Option<String>>, u64>,
```

I'd suggest making the counter hashmap use an `Option` so that, if the user didn't specify a `group_by`, you could use `None` to lookup the counter. This would avoid the need to have a "default" group with a name that might collide.